common-pg/src/main/scala/com/socrata/pg/soql/SoQLAnalysisSqlizer.scala

```diff
@@ -173,15 +173,44 @@ object SoQLAnalysisSqlizer extends Sqlizer[AnalysisTarget] {
           PostgresUniverseCommon.searchVector(primaryTableReps, Some(ctx))
       }
 
+      val searchNumberLitOpt =
+        try {
+          val number = BigDecimal.apply(searchLit.value)
+          Some(NumberLiteral(number, SoQLNumber.t)(searchLit.position))
+        } catch {
+          case _: NumberFormatException =>
+            None
+        }
+
+      val numericCols = searchNumberLitOpt match {
+        case None => Seq.empty[String]
+        case Some(searchNumberLit) =>
+          prevAna match {
+            case Some(a) =>
+              PostgresUniverseCommon.searchNumericVector(a.selection, Some(ctx))
+            case None =>
+              val primaryTableReps = rep.filter{ case (k, v) => k.qualifier.isEmpty}.values.toSeq
+              PostgresUniverseCommon.searchNumericVector(primaryTableReps, Some(ctx))
+          }
+      }
+
+      val searchVectorSql = searchVector.map {sv => s"$sv @@ plainto_tsquery('english', $searchSql)" }.toSeq
+
+      val searchPlusNumericParametricSql = numericCols.foldLeft(ParametricSql(searchVectorSql, searchSetParams)) { (acc, x) =>
+        val ParametricSql(Seq(prev), setParamsBefore) = acc
+        val ParametricSql(Seq(searchSql), setParamsAfter) =
+          Sqlizer.sql(searchNumberLitOpt.get)(repMinusComplexJoinTable, typeRep, setParamsBefore, ctx + (SoqlPart -> SoqlSearch), escape)
+        ParametricSql(Seq(s"$prev OR ($x = $searchSql)"), setParamsAfter)
+      }
+
       val andOrWhere = if (where.isDefined) " AND" else " WHERE"
-      searchVector match {
-        case Some(sv) =>
-          val fts = s"$andOrWhere $sv @@ plainto_tsquery('english', $searchSql)"
-          ParametricSql(Seq(fts), searchSetParams)
-        case None =>
-          ParametricSql(Seq(s"$andOrWhere false"), setParamsWhere)
+      if (searchPlusNumericParametricSql.sql.isEmpty) {
+        ParametricSql(Seq(s"$andOrWhere false"), setParamsWhere)
+      } else {
+        ParametricSql(searchPlusNumericParametricSql.sql.map { x => s"$andOrWhere ($x)" }, searchPlusNumericParametricSql.setParams)
```

Comment:
If `where` is defined will this end up generating SQL like `select * WHERE clause1 WHERE clause2 ...`?

Reply:
$x should be something like '(to_vector(.....) [ or numeric_col1 = ? or numeric_col2 = ? ...)'
$andOrWhere should be WHERE if there is no filter and filter + search becomes WHERE $x.
$andOrWhere should be AND if there is filter and filter + search becomes WHERE filter AND $x.